From: Martin Nowack <m.nowack@imperial.ac.uk>
Date: Mon, 30 Oct 2023 14:59:07 +0000
Subject: Handle check for thrown libc++ exceptions more general
Patch-mainline: no
References: llvm16

The wording changed slightly in newer versions.
Update the test case to support this.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 test/CXX/symex/libc++/nested_fail.cpp           | 2 +-
 test/CXX/symex/libc++/simple_exception_fail.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/CXX/symex/libc++/nested_fail.cpp b/test/CXX/symex/libc++/nested_fail.cpp
index 4dce0279..fe2df4c9 100644
--- a/test/CXX/symex/libc++/nested_fail.cpp
+++ b/test/CXX/symex/libc++/nested_fail.cpp
@@ -24,4 +24,4 @@ int main(int argc, char **args) {
   }
   return 0;
 }
-// CHECK: terminating with uncaught exception of type char*
+// CHECK: terminating {{.*}} uncaught exception of type char*
diff --git a/test/CXX/symex/libc++/simple_exception_fail.cpp b/test/CXX/symex/libc++/simple_exception_fail.cpp
index 793d9201..c3e295c2 100644
--- a/test/CXX/symex/libc++/simple_exception_fail.cpp
+++ b/test/CXX/symex/libc++/simple_exception_fail.cpp
@@ -11,4 +11,4 @@
 int main(int argc, char **args) {
   throw std::runtime_error("foo");
 }
-// CHECK: terminating with uncaught exception of type std::runtime_error: foo
\ No newline at end of file
+// CHECK: terminating {{.*}} uncaught exception of type std::runtime_error: foo
\ No newline at end of file
-- 
2.43.0

